canonical/loki-k8s-operator
# add release automation

### Changes
- Splits the CI into two different workflows, one for PRs and one for releases.
- Replaces the legacy library check with the one provided by charming-actions.
- Introduces charm release automation using the RC version of the charming-actions.
- Makes all git checkouts shallow to save some time while building.
- Cleans up some old cruft related to matrix builds that we don't use anymore.
- Harmonizes the job names to match our other repositories.

### Test Instructions
- Verify that the workflow passes for the PR.
- Verify that the charm behaves correctly once merged and the release workflow has completed.
